test/Microsoft.Dnx.Runtime.Tests/DependencyManagement/FrameworkReferenceResolverFacts.cs

```diff
@@ -55,17 +59,28 @@ public class FrameworkReferenceResolverFacts
         //[InlineData("net30", "Microsoft.Build.Engine", @"%WINDIR%\Microsoft.NET\Framework\v2.0.50727\Microsoft.Build.Engine.dll", "2.0.0.0")]
         //[InlineData("net35", "Microsoft.Build.Engine", @"REFASMSROOT\v3.5\Microsoft.Build.Engine.dll", "3.5.0.0")]
 
+        // Things not present in Client Profiles
+        [InlineData("net40", "System.Web", @"REFASMSROOT\.NETFramework\v4.0\System.Web.dll", "4.0.0.0")]
+        [InlineData("net40-client", "System.Web", "", "")]
         public void FrameworkResolverResolvesCorrectPaths(string shortFrameworkName, string assemblyName, string expectedPath, string expectedVersion)
         {
             var resolver = new FrameworkReferenceResolver();
 
             string actualPath;
             Version actualVersion;
-            Assert.True(resolver.TryGetAssembly(assemblyName, VersionUtility.ParseFrameworkName(shortFrameworkName), out actualPath, out actualVersion));
-            Assert.Equal(Environment.ExpandEnvironmentVariables(expectedPath).Replace("REFASMSROOT", FrameworkReferenceResolver.GetReferenceAssembliesPath()), actualPath);
+            var result = resolver.TryGetAssembly(assemblyName, VersionUtility.ParseFrameworkName(shortFrameworkName), out actualPath, out actualVersion);
 
-            // Having this be Version->Version equality caused some problems...
-            Assert.Equal(Version.Parse(expectedVersion).ToString(), actualVersion.ToString());
+            if (string.IsNullOrEmpty(expectedPath))
+            {
+                Assert.False(result);
+            }
+            else
+            {
+                Assert.True(result);
+                Assert.Equal(Environment.ExpandEnvironmentVariables(expectedPath).Replace("REFASMSROOT", FrameworkReferenceResolver.GetReferenceAssembliesPath()), actualPath);
+                // Having this be Version->Version equality caused some problems...
```

Comment:
Implicit operators caused weird behavior. This is old stuff though, not new in this PR.